src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -145,7 +144,9 @@ def update(self, sample: Sample[QuantityT]) -> None:
         # Update timestamps
         prev_newest = self._datetime_newest
         self._datetime_newest = max(self._datetime_newest, timestamp)
-        self._datetime_oldest = self._datetime_newest - self._time_range
+        self._datetime_oldest = (
+            self._datetime_newest - (len(self._buffer) - 1) * self._sampling_period
+        )
```

Comment:
but now the calculation has moved from the `init` method to the hot path.  maybe we can find a better name instead?  like `buffer_time_range`, `buffer_duration`, etc?

Reply:
or maybe `max_buffer_duration`

Reply:
I mean, it's a rather simple calculation.. it shouldn't impact the total runtime noticeable

Reply:
Maybe,  but they add up.  We are of course adding additional features that make things slower, so that we can have a nicer external interface, but here it looks like an easy performance win, however tiny,  especially because there's no async here, and can be reused for the historical pipeline, if necessary, etc.

But I'll leave it up to you.

Reply:
updated